documentation/bolt_defaults_reference.md.erb

```diff
@@ -0,0 +1,92 @@
+# `bolt-defaults.yaml` options
+
+This page lists the configuration options that are available in the `bolt-defaults.yaml` configuration file,
+which is supported in both the [system-wide and user-level configuration directories](configuring_bolt.md).
+The avaiable options allow you to configure how Bolt runs and set the default configuration for each of the
+available transports.
+
+Options that are set in a `bolt-defaults.yaml` file in the system-wide configuration directory will apply
+to all Bolt projects run on the system. Likewise, options set in a `bolt-defaults.yaml` file in the user-level
+configuration directory apply to all Bolt projects run by the user.
+
+## Top-level options
+
+These are the configuration options that can be set at the top-level of the file.
+
+<% @top_level[:options].each do |option, desc| -%>
+### `<%= option %>`
+
+<%= desc %>
+
+<% if @top_level[:defaults][option] -%>
+- **Default:** <%= @top_level[:defaults][option] %>
+<% end %>
+<% end %>
+
+## Additional options
+
+Several configuration options accept a map of configuration options.
+
+### `inventory-config`
+
+The top-level `inventory-config` option lets you to set the default transport used by targets that do
+not specify a transport in their URI or in the `inventory.yaml` file.
+
+<% @inventory[:options].each do |option, desc| -%>
+#### `<%= option %>`
+
+<%= desc %>
+
+<% if @inventory[:defaults][option] -%>
+- **Default:** <%= @inventory[:defaults][option] %>
+<% end %>
+<% end %>
+
+<% @transports[:options].each do |transport, options| -%>
+### `<%= transport %>`
+
+<%= @inventory[:options][transport] %>
+
+<% options.each do |option, data| -%>
+#### `<%= option %>`
+
+<%= data[:desc] %>
+
+<% if data[:type] -%>
+- **Type:** <%= data[:type] == TrueClass ? 'Boolean' : data[:type] %>
+<% end %>
+<% if @transports[:defaults][transport][option] -%>
+- **Default:** <%= @transports[:defaults][transport][option] %>
+<% end %>
+<% if transport == 'ssh' -%>
+- **External SSH Support:** <%= data[:external] == true ? 'Yes' : 'No' %>
+<% end %>
+<% end %>
+<% end %>
+
+## Example file
+
+```yaml
+color: true
+compile-concurrency: 8
+concurrency: 100
+format: human
+plugin_hooks:
+  puppet_library:
+    plugin: puppet_agent
+    stop_service: true
+plugins:
+  pkcs7:
+    private_key: ~/.ssh/my_key.pem
+    public_key: ~/.ssh/my_key.pub
+puppetdb:
+  server_urls: ["https://puppet.example.com:8081"]
+  cacert: /etc/puppetlabs/puppet/ssl/certs/ca.pem
+  token: ~/.puppetlabs/token
+puppetfile:
+  forge:
+    proxy: https://forge-proxy.org.com
+    baseurl: https://forge.org.com
+  proxy: https://forge-proxy.org.com
+save-rerun: true
+```
```

Comment:
Left out `inventory-config` for now, but wanted to see what thoughts were on having an example file that shows what having all options listed would look like. It's definitely verbose, but I think works better than having tons of smaller example files.